gax/src/main/java/com/google/api/gax/batching/FlowController.java

```diff
@@ -142,43 +142,65 @@ public String toString() {
 
   @Nullable private final Semaphore64 outstandingElementCount;
   @Nullable private final Semaphore64 outstandingByteCount;
-  @Nullable private final Long maxOutstandingElementCount;
-  @Nullable private final Long maxOutstandingRequestBytes;
+  @Nullable private final Long maxElementCountLimit;
+  @Nullable private final Long maxRequestBytesLimit;
+  @Nullable private final Long minElementCountLimit;
+  @Nullable private final Long minRequestBytesLimit;
+  @Nullable private Long currentElementCountLimit;
```

Comment:
Since these are used in multithreaded environment, it would be a good practice to declare them volatile. 

Also, given that these values are used only in simple context, it might be possible to resort to using non-blocking approach (and convert this to AtomicLong)

Reply:
`currentElementCountLimit` is modified and accessed in synchronized blocks (will add synchronize to the getters), so is it still necessary to make it volatile?  Or is it just to make the code clear?